src/pages/Checkout/CheckoutBetaPage.vue

```diff
@@ -180,43 +181,82 @@ export default {
 			isHovered: false,
 			activeLoginDuration: 3600,
 			lastActiveLogin: 0,
-			preCheckoutStep: ''
+			preCheckoutStep: '',
+			preValidationErrors: []
 		};
 	},
 	apollo: {
-		query: initializeCheckout,
 		// using the prefetch function form allows us to act on data before the page loads
 		preFetch(config, client) {
-			return client.query({
-				query: initializeCheckout
-			}).then(({ data }) => {
-				const totals = _get(data, 'shop.basket.totals');
-				// check for bonus credit and redirect if present
-				// IMPORTANT: THIS IS DEPENDENT ON THE CheckoutBeta Experiment
-				// TODO: remove once bonus credit functionality is added
-				// TODO: bonusAvailableTotal is reporting 0 once the credit has been removed in legacy basket
-				if (parseFloat(totals.bonusAvailableTotal) > 0) {
-					return Promise.reject({
-						path: '/basket',
-						query: {
-							kexpn: 'checkout_beta.minimal_checkout',
-							kexpv: 'a'
-						}
-					});
-				}
-				return data;
+			// prefetch mutation
+			// when we prefetch this mutation, any errors are listed as an array of objects
+			// HOWEVER, this doesnot go into the apollo cache so is not available on subsequent calls
+			// TODO: if errors exist in the result, mutate them into @client state so they are available
```

Comment:
@emuvente As I state here, the result of the mutation when done using prefetch is only available in this context. It's not cached so the next call lower in the code doesn't get any errors.

I see 2 options:
1. Move the validation mutation out of prefetch, updating the basket state as necessary.
2. Somehow inject the initial result into @client state, then query that @client state instead of call the mutation again.

thoughts?

Reply:
It might make sense here to do the mutation in the `beforeRouteEnter` hook for this page component, using the callback provided to `next` to use the mutation result in the initialized component (see https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards)